### PR TITLE
perf: Reduce CURRENT_TIMESTAMP queries

### PR DESCRIPTION
**Pull request by TrevorBergeron**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: bigframes/session/time.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import threading
+import time
+from typing import cast, Optional
+
+import google.cloud.bigquery as bigquery
+
+MIN_RESYNC_SECONDS = 100
+
+
+class BigquerySyncedClock:
```
Comment (tswast) [inline bigframes/session/time.py:25]: Nit: BigQuery (capital Q)

Comment (tswast) [reply bigframes/session/time.py:25]: Also, could you make some unit tests for this functionality using [freezegun](https://github.com/spulec/freezegun) to mock out the system monotonic time? Thanks!

Comment (TrevorBergeron) [reply bigframes/session/time.py]: added unit tests in new revision

**Review by tswast — APPROVED** [review]
Clever! This should really help things out.

Note: I think the benchmarks use a separate process per query if I remember correctly. If so, we might need to update those to take full advantage of this. Might even be worth some sort of "warm up" query before running the benchmarks proper.

**Pull request merged**